### DISCUSSION TITLE
Handling location for calendar

### DISCUSSION THREAD

**Discussion by Wetzel402** [home-assistant/architecture#1119] [Entity Models]
## Proposal 

**HA Core:**
- Add `location` to `local_calendar` component as string
- `location` currently exists in `calendar` as a string

**HA Frontend:**
- Add `location` field to create/edit event dialog 
- Improve event details dialog and add `location`

## Screenshots
![image](https://github.com/user-attachments/assets/51d9ee76-73e4-4ebd-993b-ec20587479eb)
![image](https://github.com/user-attachments/assets/e2cf580f-33cb-432f-becf-f1f8de45f133)

## Use case
- Add location for events

## Architecture 
- Per @allenporter comment...
  - Do we use only a string field as `calendar` currently contains?
  - Do we add `geo` to `calendar` and then add that to the associated calendar components?

## Future
- Implement import of ics for `local_calendar`

## Other considerations 
- If `geo` is added something like Google Calendar could harness and use it but how does the `frontend` expose it and make use of it?
- If `geo` is added how is it implemented for `local_calendar`?
   - Can/should the `frontend` do a Google Maps search or similar based on the `location` string and then update the `geo`?
   - Should `geo` be exposed as a field in the frontend?  If auto-filled based on a Google Maps lookup it could still be edited by the user should Maps return the wrong/incorrect location.  

Draft PRs...
https://github.com/home-assistant/core/pull/122561
https://github.com/home-assistant/frontend/pull/21471

**Comment by allenporter** [comment]
I have two initial big picture feedback points on the proposal, then some more detail.
(1) Let's  separate discussions about location and people since they are in very different states. 
(2) The use cases discussion is still quite minimal and needs to actually talk about real world use cases we're considering here to drive implementation requirements.  We don't need to get into the work plan for specific changes that are needed yet, but more focus on (a) multiple key user use cases (b) semantics/constraints/guarantees of proposed APIs and therefore the implications for both calendar integrations and other parts of home assistant

Happy to talk more in our discord discussion.

## Location

My impression is that location is already part of the entity model as a free form text string (#885) and supported by multiple integrations, so this does not really need an architecture discussion. I think the question is how to represent in the frontend?  I believe it needs:
- a plan for how calendar event locations will be represented on the calendar pages
- how the frontend will deal with various free from text.

So that brings in some other questions depending on the use case... Given a user interface being exposed, we might expect integration with map cards, geolocation platforms or map routing integrations, etc so then the interesting architecture questions come up.  (Though the rfc5545 also has a separate "GEO" field that is a lat lng which I think we may not really be talking about here, or we (ab)use location by sticking lat longs in it).  A bigger picture view of where we're going this could help once we start exposing these details to users, otherwise its not clear what you're supposed to do with it.

## People

You're going to need to  further expand on the use cases for people quite a bit.  This is going to get complex (more than is specified here) so its going to need significant motivation to be accepted.

Once we have more use cases settled, attendees is still underspecified here. Some thoughts are:
- How will these attendees integrate in the rest of home assistant? For example: used in automations, or how does it integrate with other ways things are associated with people like people + locations
- You seem to be proposing an attendee object with a single field, but i can't tell, so we'll want to discuss what that object is. [Attendees](https://www.rfc-editor.org/rfc/rfc5545#section-3.8.4.1) are a very rich "object" with lots of fields so what's in and whats out now and going forward
- More on the semantics of what these fields mean in terms of home assistant -- i'm not understanding if the proposal is exact name matching, or entities, or what.
- More conviction about how multiple integrations can integrate with this in an understandable way (e.g. how are non-person attendees handled).  We shouldn't punt on having a credible plan on how multiple integrations will support this.

**Comment by Wetzel402** [comment]
`Attendee` discussion moved...
https://github.com/home-assistant/architecture/discussions/1099

This thread will now focus on `location`.
